internal/providers/terraform/azure/cosmosdb_cassandra_keyspace.go

```diff
@@ -47,6 +47,22 @@ func cosmosDBCostComponents(d *schema.ResourceData, u *schema.UsageData, account
 	region := lookupRegion(account, []string{"account_name", "resource_group_name"})
 	geoLocations := account.Get("geo_location").Array()
 
+	// The geo_location attribute is a required attribute however it can be an empty list because of
+	// expressions evaluating as nil, e.g. using a data block. If the geoLocations variable is empty
+	// we set it as a sane default which is using the location from the parent region.
+	if len(geoLocations) == 0 {
+		log.WithFields(log.Fields{
+			"resource": d.Address,
+		}).Debugf("empty set of geo_location attributes provided using fallback region %s", region)
+
+		geoLocations = []gjson.Result{
```

Comment:
Seems odd we're creating a gjson, but I guess it will involve more refactoring at the moment to pass the location directly to the functions we are calling here?

Reply:
yeah the `geo_locations` are used by a ton of downstream functions that expect an array of `gjson.Result` it's jankey, but I didn't think it warranted a ton of refactoring

Reply:
👍